### PR TITLE
Minor fixes in AnyValue and scons

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

- ensure that a YAML field: `some-field: '12345'` is read as a string by the C++ parser (which currently returns an integer)
- make C++ parser consistent with `ruamel.yaml`
- Also: minor formatting update in scons output (mixed windows/linux style `os.path.sep`s in install message on windows)

This is a minor fix that is left over after closing #881. Also cherry-picking minor fixes from #892 to ensure that fixes of existing features survive.

**Checklist**

- [x] There is a clear use-case for this code change
- [x] The commit message has a short title & references relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] The pull request is ready for review

### CODE REVIEW

File: src/base/AnyMap.cpp [hunk 1]
```diff
@@ -190,7 +190,11 @@ struct convert<Cantera::AnyValue> {
         if (node.IsScalar()) {
             // Scalar nodes are int, doubles, or strings
             std::string nodestr = node.as<std::string>();
-            if (isInt(nodestr)) {
+            if (node.Tag() == "!") {
+                // Undo conversion from string to other formats (done
+                // automatically by yaml-cpp, e.g. '12345' is converted to int)
+                target = nodestr;
+            } else if (isInt(nodestr)) {
```
Comment (bryanwweber) [inline src/base/AnyMap.cpp]: I believe this change will allow resolution of this comment in `ctml2yaml.py`: https://github.com/Cantera/cantera/blob/773523a4e314e4ce4bfec379b438c5ca55cebf5c/interfaces/cython/cantera/ctml2yaml.py#L1947-L1950 🎉  (I'm not suggesting that be added to this PR)

Comment (ischoegl) [reply src/base/AnyMap.cpp]: Ok. Happy that it may help elsewhere!

File: src/base/AnyMap.cpp [hunk 2]
```diff
@@ -190,7 +190,11 @@ struct convert<Cantera::AnyValue> {
         if (node.IsScalar()) {
             // Scalar nodes are int, doubles, or strings
             std::string nodestr = node.as<std::string>();
-            if (isInt(nodestr)) {
+            if (node.Tag() == "!") {
+                // Undo conversion from string to other formats (done
+                // automatically by yaml-cpp, e.g. '12345' is converted to int)
```
Comment (speth) [inline src/base/AnyMap.cpp]: I think the fix you've added here, at least as far as I understand `yaml-cpp`, makes sense, but the comment is not correct -- the automatic conversions are being done here in this method, based on whether the string value looks like an integer or floating point value.

**Review by speth — COMMENTED** [review]
Looks basically fine to me. Could you update the commit "[scons] Small updates/fixes" to have a more descriptive message? It's not immediately obvious what the purpose of these changes is.

File: src/base/AnyMap.cpp (hunk 2 above)
Comment (ischoegl) [reply src/base/AnyMap.cpp]: `yaml-cpp` erroneously strips the `'` from the string (which causes the issue), so yes, this needs updating.

**Comment by ischoegl** [comment]
Sorry for pushing again, but I noticed that the note in `ch4_ion.yaml` was not indented correctly (!). I stumbled over this when I tried to verify that the missing quotes I corrected here weren't an artifact produced by `cti2yaml` and/or `ctml2yaml` as the note happens to be a valid numeric value (result: both converters create the quotes, and indent correctly).

**Comment by ischoegl** [comment]
Will probably add another update for the formatting, as I just found a snippet for Anaconda on windows documenting the status before this PR
```
[...]
Cantera has been successfully installed.

File locations:

  applications                C:/Users/<user>/cantera\bin
  library files               C:/Users/<user>/cantera\lib
  C++ headers                 C:/Users/<user>/cantera\include
  samples                     C:/Users/<user>/cantera\samples
  data files                  C:/Users/<user>/cantera\data
  Python package (cantera)    C:\Users\<user>\anaconda3\envs\cantera-dev\Lib\site-packages
  Python samples              C:\Users\<user>\anaconda3\envs\cantera-dev\Lib\site-packages\cantera\examples
  Matlab toolbox              C:/Users/<user>/cantera\matlab\toolbox
  Matlab samples              C:/Users/<user>/cantera\samples\matlab

An m-file to set the correct matlab path for Cantera is at:

  C:/Users/<user>/cantera\matlab\toolbox\ctpath.m

scons: done building targets.
```
All this takes is a couple of `os.path.normpath`s ...

**Comment by ischoegl** [comment]
:tada: ... better:
```
[...]
Cantera has been successfully installed.

File locations:

  applications                C:\Users\ischo\cantera\bin
  library files               C:\Users\ischo\cantera\lib
  C++ headers                 C:\Users\ischo\cantera\include
  samples                     C:\Users\ischo\cantera\samples
  data files                  C:\Users\ischo\cantera\data
  Python package (cantera)    C:\Users\ischo\miniconda3\envs\cantera-dev\Lib\site-packages
  Python samples              C:\Users\ischo\miniconda3\envs\cantera-dev\Lib\site-packages\cantera\examples
  Matlab toolbox              C:\Users\ischo\cantera\matlab\toolbox
  Matlab samples              C:\Users\ischo\cantera\samples\matlab

An m-file to set the correct matlab path for Cantera is at:

  C:\Users\ischo\cantera\matlab\toolbox\ctpath.m

scons: done building targets.
```

File: src/base/AnyMap.cpp (hunk 2 above)
Comment (speth) [reply src/base/AnyMap.cpp]: I don't think it's correct to say that what `yaml-cpp` is doing is erroneous. If I have the YAML `{a: 12345, b: '12345', c: hello, d: 'hello'}`, then the string values of (a) and (b) should be equal, the same as the string values of (c) and (d) are equal. The only distinction is that (b) and (d) were indicated as being strings by being quoted, which is captured in the value of the "tag". The comment I would suggest is just "Prevent quoted strings from being implicitly converted to numeric types".

Comment (ischoegl) [reply src/base/AnyMap.cpp]: Correct; shouldn’t have inferred as much as I did from `ruamel.yaml`; `yaml-cpp` clearly works at a lower level. Resolving as
```
                // Prevent quoted strings from being implicitly converted to
                // numeric types, e.g. the quoted YAML string '12345' should not
                // be interpreted as an integer
```

**Comment by ischoegl** [comment]
@speth ... I think I took care of the requested change of the comment. If it’s adequate, I think this should be ready to merge.

**Review by speth — APPROVED** [review]
Looks good. Sorry for the delayed review.

**Comment by ischoegl** [comment]
NP - I figure you took a well-deserved break :wink:

**Pull request merged**